airflow/providers/amazon/aws/transfers/s3_to_sql.py

```diff
@@ -0,0 +1,158 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import os
+from typing import TYPE_CHECKING, Any, Sequence
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base import BaseHook
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.providers.common.sql.hooks.sql import DbApiHook
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+from typing_extensions import Literal
+
+try:
+    import csv as csv
+except ImportError as e:
+    from airflow.exceptions import AirflowOptionalProviderFeatureException
+
+    raise AirflowOptionalProviderFeatureException from e
+
+
+class S3ToSqlOperator(BaseOperator):
+    """
+        Loads Data from S3 into a SQL Database.
+        Data should be readable as CSV.
+
+        This operator downloads a file from an S3, reads it via `csv.reader`
+        and inserts the data into a SQL database using `insert_rows` method.
+        All SQL hooks are supported, as long as it is of type DbApiHook
```

Comment:
Hmmm... Operator limited by CSV sources?

Reply:
Agree with this. Either you explicitly specify in the operator name that this operator reads only CSV, either you can make this operator generic by adding a new parameter `parser` which would be a function responsible of reading the input from the source. The actual processing of CSV in your would be handled by this new function `parser`. Between the 2 solutions, I prefer the second one. My 2 cents

Reply:
@vincbeck I agree. I like the Idea of using a `parser` parameter. Do you think it makes sense to use a default parser?

Reply:
I dont think it makes sense as default but I do think that providing at least one parser in code or documentation makes sense. A CSV parser would be excellent!

Reply:
Hi @vincbeck, I have changed the operator to use a parser instead and have documented it accordingly. 
Could you please have a and let me know what you think about it

Reply:
It looks really good! Thanks for making the changes. I added some comments but they are mostly nitpicks and minor comments. Good job!